exasol_sphinx_github_pages_generator/generate_index.py

```diff
@@ -0,0 +1,157 @@
+    for release in release_list:
+        if release == "_sources" or release == "_static":
+            continue
+        if release != source_branch:
+            release_list_dicts.append({"release": release,
+                                       "release_path": f"{find_index(Path('.'), release)}"})
+    return release_list_dicts
+
+
+def get_releases(target_branch: str, target_branch_exists_remote: bool, source_branch: str, target_worktree: Path) \
+        -> List[Dict[str, str]]:
+    """
+    Find all releases already documented in target_branch, and return them tih the path pointing to their index.html
+    file. Additionally also return the current release and index.html-file-path. The current release is titled "latest".
+    Found releases are returned as a list of dictionaries.
+    In order to find the releases in the target_branch, it is checked out in an additional temporary worktree.
+    Takes the root-directory name for each release and returns it. Gives bad results if directory names contain slashes.
+    Please remove slashes from directory names before using them to generate the root directory for your
+    release-documentation.
+
+    :param target_branch: The branch the generated documentation should end up on.
+    :param target_branch_exists_remote: Bool representing whether the target_branch already exist in the remote
+        repository.
+    :param source_branch: The branch we are currently generating the documentation for.
+    :param target_worktree: Worktree/path all generated project documentation is put into.
+    :return: List of dictionaries containing the release name and path to its index.html file.
+    """
+    if target_branch_exists_remote:
+        find_index_worktree_path = Path(os.getcwd()) / "target-branch-for-index/"
+        completed = run(["git", "worktree", "add", find_index_worktree_path, target_branch, "--force"])
+        if completed.returncode != 0:
+            sys.exit(inspect.cleandoc(f"""checking out target_branch {target_branch} failed, although given 
+                     'target_branch_exists_remote' was 'True'. Check if target_branch really exists on remote?
+                     received Error:
+                        returncode: {completed.returncode},
+                        stderr: {completed.stderr},
+                        stdout: {completed.stdout}"""
+                     ))
+        with change_and_restore(find_index_worktree_path):
+            current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True,
+                                 check=True)
+            print(f"current_branch {current_branch.stdout} in find_index_worktree_path {find_index_worktree_path}.")
+            # does not work if release branch name contains slash
+            release_list = (name for name in os.listdir(find_index_worktree_path)
+                            if os.path.isdir(os.path.join(find_index_worktree_path, name)))
```

Comment:
we need to make this safe. Except for our main branch, all branches follow the format label_type/issue_number_issue_summary.
It is true in the CI this we shouldn't have slashes, but in all other cases.
The question is, where do create the sphinx docu for a branch with slashes? If I see this correctly, you replace a slash with dash (-). This means there shouldn't be no directories with slashes and if would find one, we can ignore it.

Reply:
this is an old comment. i will remove it. you are right, all relevant slashes are replaced with "-" before this point.